### DISCUSSION TITLE
Are rich comparisons (PEP 207) implemented?

### DISCUSSION THREAD

**Discussion by dbischof90** [pythonnet/pythonnet#2391] [Q&A]
Hi,

I’m currently evaluating whether using Python.NET can yield an ergonomic solution for a problem at work. 

I’m essentially dealing with some form of an array type and would like to do operations that are comparable to numpy, e.g.

```python
c = a > b
assert type(c) == “MyArrayType”
```

I can’t really find a conclusive answer to that in the source code - you mention “rich comparisons” in a few type definitions, however your tests (`TestOperator.cs`) only cover simple comparisons. 

Is it possible to implement those in the client code?

**Comment by lostmsu** [comment]
Can you please clarify that you imagine should be possible? Do you want a custom C# type that has a comparison operation which returns non-`bool`?

**Reply by dbischof90** [reply]
Hi! Yes, essentially what I described in my snippet. For my array type, if `b` was a number type, it would be great if I could return my array type with boolean values at which indices my array is larger than that number - as mentioned, as a good example on how that would/should work I would point to numpy. 


I assume that isn't working yet?

**Reply by lostmsu** [reply]
I have not tried it, but you can try defining an instance member just like you would in Python. E.g. `int[] __lt__(int[] other) => ...`

**Reply by dbischof90** [reply]
Ah so that method is overridden? I haven't looked too deeply into the library - so it's not some interface method but just a plain old method? So it's not going to be an issue that this method returns something else than `PyBool`?

I'll try to whip up a test case and try that, thanks.
